beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -0,0 +1,169 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/go-ssz"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/roughtime"
+	"github.com/prysmaticlabs/prysm/shared/traceutil"
+	"go.opencensus.io/trace"
+)
+
+// validateAggregateAndProof verifies the aggregated signature and the selection proof is valid before forwarding to the
+// network and downstream services.
+func (r *Service) validateAggregateAndProof(ctx context.Context, pid peer.ID, msg *pubsub.Message) bool {
+	if pid == r.p2p.PeerID() {
+		return true
+	}
+
+	ctx, span := trace.StartSpan(ctx, "sync.validateAggregateAndProof")
+	defer span.End()
+
+	// To process the following it requires the recent blocks to be present in the database, so we'll skip
+	// validating or processing aggregated attestations until fully synced.
+	if r.initialSync.Syncing() {
+		return false
+	}
+
+	raw, err := r.decodePubsubMessage(msg)
+	if err != nil {
+		log.WithError(err).Error("Failed to decode message")
+		traceutil.AnnotateError(span, err)
+		return false
+	}
+	m, ok := raw.(*ethpb.AggregateAttestationAndProof)
+	if !ok {
+		return false
+	}
+
+	attSlot := m.Aggregate.Data.Slot
+
+	// Verify aggregate attestation has not already been seen via aggregate gossip, within a block, or through the creation locally.
+	// TODO(3835): Blocked by operation pool redesign
```

Comment:
Maybe we can open a specific tracking issue for this item instead of just relying on #3835

Reply:
I'll address this today